src/org/jitsi/impl/neomedia/transform/fec/FlexFec03Receiver.java

```diff
@@ -0,0 +1,341 @@
+            {
+                logger.error("Given FlexFEC packet is too small");
+                return false;
+            }
+            if (recoveredPacket.getBuffer().length < RawPacket.FIXED_HEADER_SIZE)
+            {
+                logger.error("Given RawPacket buffer is too small");
+                return false;
+            }
+            // Copy over the recovery RTP header data from the fec packet
+            // (fecPacket contains the RTP header, so we need to copy from it
+            // starting after that)
+            System.arraycopy(fecPacket.getBuffer(), fecPacket.getFlexFecHeaderOffset(),
+                recoveredPacket.getBuffer(), 0, RawPacket.FIXED_HEADER_SIZE);
+
+            // Copy over the recovery rtp payload data from the fec packet
+            System.arraycopy(
+                fecPacket.getBuffer(),
+                fecPacket.getFlexFecHeaderOffset() + fecPacket.getFlexFecHeaderSize(),
+                recoveredPacket.getBuffer(),
+                RawPacket.FIXED_HEADER_SIZE,
+                fecPacket.getFlexFecPayloadLength());
+
+            return true;
+        }
+
+        /**
+         * Xor the RTP headers of source and destination
+         * @param source the packet to xor the header from
+         * @param dest the packet to xor the header into
+         */
+        private void xorHeaders(RawPacket source, RawPacket dest)
+        {
+            // XOR the first 2 bytes of the header: V, P, X, CC, M, PT fields.
+            dest.getBuffer()[0] ^= source.getBuffer()[0];
+            dest.getBuffer()[1] ^= source.getBuffer()[1];
+
+            // XOR the length recovery field.
+            int length = (source.getLength() & 0xffff) - RawPacket.FIXED_HEADER_SIZE;
+            dest.getBuffer()[2] ^= (length >> 8);
+            dest.getBuffer()[3] ^= (length & 0x00ff);
+
+            // XOR the 5th to 8th bytes of the header: the timestamp field.
+            dest.getBuffer()[4] ^= source.getBuffer()[4];
+            dest.getBuffer()[5] ^= source.getBuffer()[5];
+            dest.getBuffer()[6] ^= source.getBuffer()[6];
+            dest.getBuffer()[7] ^= source.getBuffer()[7];
+
+            // Skip the 9th to 12th bytes of the header.
+        }
+
+        /**
+         * Xor the payloads of the source and destination
+         * @param source the packet to xor the payload from
+         * @param sourceOffset the offset in the source packet at which the
+         * payload begins.  Note that this is not necessarily the location
+         * at which the RTP payload starts...for the purpose of FlexFEC,
+         * everything after the fixed RTP header is considered the 'payload'
+         * @param dest the packet to xor the payload into
+         * @param destOffset the offset in the dest packet at which the payload
+         * begins
+         * @param payloadLength the length of the source's payload
+         */
+        private void xorPayloads(RawPacket source, int sourceOffset,
```

Comment:
If you're passing offsets anyway, might as well pass the byte[] instead of RawPacket

Reply:
sure, fixed